src/commands/aksInspektorGadget/aksInspektorGadget.ts

```diff
@@ -0,0 +1,173 @@
+    _context: IActionContext,
+    target: any
+): Promise<void> {
+    await gadgetDeployUndeploy(_context, target, InspektorGadget.Undeploy);
+}
+
+async function gadgetDeployUndeploy(
+    _context: IActionContext,
+    target: any,
+    gadget: InspektorGadget
+): Promise<void> {
+    const kubectl = await k8s.extension.kubectl.v1;
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    const clusterExplorer = await k8s.extension.clusterExplorer.v1;
+
+
+    if (!kubectl.available) {
+        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+        return undefined;
+    }
+
+    if (!cloudExplorer.available) {
+        vscode.window.showWarningMessage(`Cloud explorer is unavailable.`);
+        return undefined;
+    }
+
+    if (!clusterExplorer.available) {
+        vscode.window.showWarningMessage(`Cluster explorer is unavailable.`);
+        return undefined;
+    }
+
+    const clusterInfo = await getKubernetesClusterInfo(target, cloudExplorer, clusterExplorer);
+    if (failed(clusterInfo)) {
+        vscode.window.showErrorMessage(clusterInfo.error);
+        return undefined;
+    }
+
+    await prepareInspektorGadgetInstall(clusterInfo.result, gadget, clusterInfo.result.kubeconfigYaml, kubectl);
+}
+
+async function prepareInspektorGadgetInstall(
+    cloudTarget: KubernetesClusterInfo,
+    gadget: InspektorGadget,
+    kubeconfig: string,
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>
+): Promise<void> {
+    const clustername = cloudTarget.name;
+
+    switch (gadget) {
+        case InspektorGadget.Deploy:
+            await deployGadget(clustername, kubeconfig, kubectl);
+            return;
+        case InspektorGadget.Undeploy:
+            const answer = await vscode.window.showInformationMessage(`Do you want to undeploy gadget in selected cluster?`, "Yes", "No");
+            if (answer === "Yes") {
+                await unDeployGadget(clustername, kubeconfig, kubectl);
+            }
+            return;
+    }
+}
+
+async function deployGadget(
+    clustername: string,
+    clusterConfig: string,
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>) {
+    const command = "deploy";
+
+    return await runKubectlGadgetCommands(clustername, command, clusterConfig, kubectl);
+}
+
+async function unDeployGadget(
+    clustername: string,
+    clusterConfig: string,
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>) {
+    const command = "undeploy";
+
+    return await runKubectlGadgetCommands(clustername, command, clusterConfig, kubectl);
+}
+
+async function runKubectlGadgetCommands(
+    clustername: string,
+    command: string,
+    clusterConfig: string,
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>) {
+
+    const kubectlGadgetPath = await getKubectlGadgetBinaryPath();
+    let kubetlGadgetBinaryPath = "";
+
+    if (failed(kubectlGadgetPath)) {
+        vscode.window.showWarningMessage(`Gadget path is not found ${kubectlGadgetPath.error}`);
+    }
+
+    if (kubectlGadgetPath.succeeded) {
```

Comment:
Just to reduce the number of path variables in scope, I'd favour something like:
```javascript
const  binaryPathDir = path.dirname(kubectlGadgetPath.result);
```